src/components/header/OverflowEdgeIndicator.js

```diff
@@ -2,20 +2,34 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+// @flow
+
 import React, { PureComponent, PropTypes } from 'react';
 import classNames from 'classnames';
 
 import './OverflowEdgeIndicator.css';
 
+type Props = {
+  className: string,
+  children: number,
+};
+
 class OverflowEdgeIndicator extends PureComponent {
-  constructor(props) {
+  props: Props;
+  state = {
+    overflowsOnTop: false,
+    overflowsOnRight: false,
+    overflowsOnBottom: false,
+    overflowsOnLeft: false,
+  };
+  _containerCreated: (elem: HTMLDivElement) => void;
+  _container: HTMLDivElement;
+  _contentsWrapperCreated: (elem: HTMLDivElement) => void;
+  _contentsWrapper: HTMLDivElement;
+  _onScroll: () => void;
```

Comment:
I saw some examples of that in the code, but when I saw the `any` and the suggestion that it should be avoided I tried this way. Thanks for the explanation why its used, I'll change it to use `(this: any)`.